include/Audio-player/MusicLibrary.hpp

```diff
@@ -1,49 +1,62 @@
-#ifndef MUSICLIBRARY_MUSICLIBRARY_HPP
-#define MUSICLIBRARY_MUSICLIBRARY_HPP
+#ifndef MUSICLIBRARY_INCLUDE_AUDIO_PLAYER_MUSICLIBRARY_HPP
+#define MUSICLIBRARY_INCLUDE_AUDIO_PLAYER_MUSICLIBRARY_HPP
 
-#include <vector>
-#include <iostream>
-#include <algorithm>
-#include <type_traits>
 #include <filesystem>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
 
+#include "Album.hpp"
+#include "FileManager.hpp"
 #include "Playlist.hpp"
 #include "Song.hpp"
-#include "Album.hpp"
 
 namespace fs = std::filesystem;
 
 class MusicLibrary {
-    std::vector<Playlist> playlists;
-    std::vector<Song> songs;
-    std::vector<Album> albums;
-    fs::path directory;
-
-public:
+  public:
     MusicLibrary() = default;
+
     MusicLibrary(fs::path _directory);
-    void addSong(const Song& song);
-    void addPlaylist(const Playlist& playlist);
 
-    template <typename T>
-    void printVector(const std::vector<T> &vec);
+    void addPlaylist(const Playlist &playlist);
 
-//    template <typename T>
-//    void removeItem(const std::string &itemTitle);
+    template <typename T> void printVector(const std::vector<T> &vec);
 
-    template <typename T>
-    std::vector<T>& getVector();
+    template <typename T> bool isEmpty(std::vector<T> &vector);
 
-    template <typename T>
-    bool isEmpty(std::vector<T> &vector);
+    void updateSongs(FileManager &fm);
+
+    void addSongToAlbum(
+        const std::string &albumName, const std::shared_ptr<Song> &song);
+
+    void updateAlbums();
+
+    const std::unordered_map<std::string, Album> &getAlbumsMap() const {
```

Comment:
It's better not to add implementetions to methods in .hpp, but if you do be consistent with it. getSong() has implementetion in .cpp but all the others doesn't so it's not consistent.